File: rust/src/leetcode/Q217_contains_duplicate.rs
/*
 * @lc app=leetcode id=217 lang=rust
 *
 * [217] Contains Duplicate
 */

// @lc code=start
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen: HashSet<i32> = HashSet::new();

        for n in nums {
            if !seen.insert(n) {
                return true;
            }
        }

        false
    }
}
// @lc code=end

File: python/Q217_contains_duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#


# @lc code=start
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        seen: set[int] = set()

        for n in nums:
            if n in seen:
                return True

            seen.add(n)

        return False


# @lc code=end
